From: Thomas Weißschuh <thomas.weissschuh@linutronix.de>
Subject: compiler.h: remove ARCH_SEL()
Date: Mon, 13 Oct 2025 11:12:02 +0200

Its last user was removed in commit 8ea815399c3f ("compiler: remove
__ADDRESSABLE_ASM{_STR,}() again").

Link: https://lkml.kernel.org/r/20251013-arch-sel-v1-1-7eef9b22ceb0@linutronix.de
Signed-off-by: Thomas Weißschuh <thomas.weissschuh@linutronix.de>
Cc: Luc Van Oostenryck <luc.vanoostenryck@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/compiler.h |    6 ------
 1 file changed, 6 deletions(-)

--- a/include/linux/compiler.h~compiler-remove-arch_sel
+++ a/include/linux/compiler.h
@@ -269,12 +269,6 @@ static inline void *offset_to_ptr(const
 
 #endif /* __ASSEMBLY__ */
 
-#ifdef CONFIG_64BIT
-#define ARCH_SEL(a,b) a
-#else
-#define ARCH_SEL(a,b) b
-#endif
-
 /*
  * Force the compiler to emit 'sym' as a symbol, so that we can reference
  * it from inline assembler. Necessary in case 'sym' could be inlined
_
